src/odin/enhancedtrippath.cc

```diff
@@ -468,18 +471,122 @@ bool EnhancedTripLeg_Edge::HasNonDirectionalTurnLane() const {
   return false;
 }
 
-bool EnhancedTripLeg_Edge::ActivateTurnLanes(uint16_t turn_lane_direction) {
-  bool activated = false;
-  for (auto& turn_lane : *(mutable_turn_lanes())) {
+bool EnhancedTripLeg_Edge::HasTurnLane(uint16_t turn_lane_direction) const {
+  for (const auto& turn_lane : turn_lanes()) {
+    // Return true if the specified turn lane exists
     if (turn_lane.directions_mask() & turn_lane_direction) {
-      turn_lane.set_is_active(true);
-      activated = true;
+      return true;
     }
   }
-  return activated;
+  return false;
+}
+
+uint16_t EnhancedTripLeg_Edge::ActivateTurnLanesFromLeft(uint16_t turn_lane_direction,
+                                                         uint16_t activated_max) {
+  uint16_t activated_count = 0;
+  // Make sure turn lane has a direction
+  if (!HasNonDirectionalTurnLane()) {
+    for (auto& turn_lane : *(mutable_turn_lanes())) {
+      // Stop processing the lanes if the activated maximum has been reached
+      if (activated_count >= activated_max) {
+        break;
+      }
+
+      // If the turn lane is in the specified direction then activate the lane
+      // and increment the activated count
+      if (turn_lane.directions_mask() & turn_lane_direction) {
+        turn_lane.set_is_active(true);
+        ++activated_count;
+      }
+    }
+  }
+  return activated_count;
+}
+
+uint16_t EnhancedTripLeg_Edge::ActivateTurnLanesFromRight(uint16_t turn_lane_direction,
+                                                          uint16_t activated_max) {
+  uint16_t activated_count = 0;
+  // Make sure turn lane has a direction
+  if (!HasNonDirectionalTurnLane()) {
+    for (auto turn_lane_iter = mutable_turn_lanes()->rbegin();
+         turn_lane_iter != mutable_turn_lanes()->rend(); ++turn_lane_iter) {
+      //    for (auto& turn_lane : *(mutable_turn_lanes())) {
+      // Stop processing the lanes if the activated maximum has been reached
+      if (activated_count >= activated_max) {
+        break;
+      }
+
+      // If the turn lane is in the specified direction then activate the lane
+      // and increment the activated count
+      if (turn_lane_iter->directions_mask() & turn_lane_direction) {
+        turn_lane_iter->set_is_active(true);
+        ++activated_count;
+      }
+    }
+  }
+  return activated_count;
+}
+
+uint16_t
+EnhancedTripLeg_Edge::ActivateTurnLanes(uint16_t turn_lane_direction,
```

Comment:
determines which lanes should be activated based on intersection, current maneuver, and next maneuver